index.js

```diff
@@ -46,6 +46,40 @@ function runNightwatch (done) {
 
 	try {
 		Nightwatch.cli(function (argv) {
+			// Set app-specific env for nightwatch session
+
+			// If possible, argv inputs and environment variables will be merged together
+			// If not, argv inputs will override environment variables.
+			if (process.env.KNE_TEST_PATHS && argv.test_paths) {
+				process.env.KNE_TEST_PATHS += ',' + argv.test_paths;
+			} else if (process.env.KNE_TEST_PATHS === undefined && argv.test_paths) {
+				process.env.KNE_TEST_PATHS = argv.test_paths;
+			} else if (process.env.KNE_TEST_PATHS === undefined && argv.test_paths === undefined) {
+				console.log('\nNo test paths provided. Either set the --test_paths config option or the KNE_TEST_PATHS environment variable');
+				done();
+			}
+			process.env.KNE_SELENIUM_SERVER = selenium.path;
+			if (process.env.KNE_PAGE_OBJECT_PATHS) {
+				process.env.KNE_PAGE_OBJECT_PATHS += ',' + path.resolve(__dirname, 'lib/src/pageObjects/');
+			} else {
+				process.env.KNE_PAGE_OBJECT_PATHS = path.resolve(__dirname, 'lib/src/pageObjects/');
+			}
+			if (argv.po_paths !== undefined) {
+				process.env.KNE_PAGE_OBJECT_PATHS += ',' + argv.popages;
+			}
+			if (process.env.KNE_EXCLUDE_TEST_PATHS === undefined) {
+				process.env.KNE_EXCLUDE_TEST_PATHS = ''; // Needs to be initialised for .split in conf.js file.
+			} else if (argv.exclude_paths !== undefined) {
+				process.env.KNE_EXCLUDE_TEST_PATHS += ',' + argv.exclude_paths;
+			}
+			if (process.env.KNE_SELENIUM_START_PROCESS === undefined) {
```

Comment:
@webteckie I disagree with this one, since KNE_SELENIUM_START_PROCESS might be `false`, in which case we shouldn't be setting it to `true`! We really do need to check for undefined here.